src/components/container/ConnectDevicesMenu/ConnectDevicesMenu.tsx

```diff
@@ -0,0 +1,229 @@
+import MyDataHelps, { DataCollectionSettings, ExternalAccount } from '@careevolution/mydatahelps-js';
+import React, { ReactElement, useEffect, useState } from 'react';
+import ConnectDevicesIcon from '../../../assets/connectdevices.svg';
+import FitbitIcon from '../../../assets/fitbit.svg';
+import GarminIcon from '../../../assets/garmin.svg';
+import AppleHealthIcon from '../../../assets/appleHealth.svg';
+import GoogleFitIcon from '../../../assets/googleFit.svg';
+import OmronIcon from '../../../assets/omron.png';
+import { Action, Title } from '../../presentational';
+import "./ConnectDevicesMenu.css"
+import { FontAwesomeSvgIcon } from 'react-fontawesome-svg-icon';
+import { faRefresh } from '@fortawesome/free-solid-svg-icons';
+import { getFitbitProviderID, getGarminProviderID, getOmronProviderID } from '../../../helpers/providerIDs';
+import { previewAccounts, previewSettings } from './ConnectDevicesMenu.previewdata';
+import language from '../../../helpers/language';
+
+export type DeviceAccountType = "Fitbit" | "Garmin" | "AppleHealth" | "GoogleFit" | "Omron";
+
+export interface ConnectDevicesMenuProps {
+    innerRef?: React.Ref<HTMLDivElement>
+    accountTypes?: DeviceAccountType[]
+    title?: string
+    text?: string
+    previewState?: "iOS" | "Android" | "Web" | "ConnectedStates"
+    headerVariant?: "large" | "medium"
+}
+
+export default function (props: ConnectDevicesMenuProps) {
+    const [loading, setLoading] = useState(true);
+    const [settings, setSettings] = useState<DataCollectionSettings | null>(null);
+    const [deviceExternalAccounts, setDeviceExternalAccounts] = useState<ExternalAccount[] | null>(null);
+    const [platform, setPlatform] = useState<string | null>(null);
+
+    function initialize() {
+        if (props.previewState) {
+            setSettings(previewSettings);
+
+            setDeviceExternalAccounts([]);
+            if (props.previewState == "ConnectedStates") {
+                setDeviceExternalAccounts(previewAccounts);
+            }
+
+            setPlatform("Web");
+            if (props.previewState == "iOS") {
+                setPlatform("iOS");
+            } else if (props.previewState == "Android") {
+                setPlatform("Android");
+            }
+            setLoading(false);
+            return;
+        }
+
+        Promise.all([MyDataHelps.getDataCollectionSettings(), MyDataHelps.getExternalAccounts(), MyDataHelps.getDeviceInfo()])
+            .then(([settings, accounts, deviceInfo]) => {
+                setSettings(settings);
+                setDeviceExternalAccounts(accounts);
+                setPlatform(deviceInfo.platform);
```

Comment:
`deviceInfo` is null when I attempted to test the component against MDH QA. Would be nice to test it locally.

Reply:
Fixed